internal/controller/resilient_deployment_workflow.go

```diff
@@ -0,0 +1,564 @@
+
+	history := GetHistory()
+
+	kObj := GetKuadrantFromTopology(topology)
+	lObj := GetLimitadorFromTopology(topology)
+	if lObj == nil {
+		logger.V(level).Info("limitador resource has not being created yet.")
+		return nil
+	}
+
+	wasConfigured := r.isConfigured(history.kuadrant)
+	nowConfigured := r.isConfigured(kObj)
+
+	if wasConfigured && !nowConfigured {
+		logger.V(level).Info("spec.storage should be removed from the limitador resource", "status", "cleanup")
+		lObj.Spec.Storage = nil
+		err := r.updateLimitador(ctx, lObj)
+		if err != nil {
+			logger.V(level).Info("failed to update limitador resource", "status", "error", "error", err)
+			return nil
+		}
+		return nil
+	}
+
+	if !nowConfigured {
+		logger.V(level).Info("CounterStorage not configured", "status", "exiting")
+		return nil
+	}
+	logger.V(level).Info("CounterStorage configured", "status", "contiune")
+
+	lObj.Spec.Storage = kObj.Spec.Resilience.CounterStorage
+	err := r.updateLimitador(ctx, lObj)
+	if err != nil {
+		logger.V(level).Info("failed to update limitador resource", "status", "error", "error", err)
+		return nil
+	}
+
+	return nil
+}
+
+func (r *ResilienceCounterStorageReconciler) isConfigured(kObj *kuadrantv1beta1.Kuadrant) bool {
+	if kObj == nil {
+		return false
+	}
+	if resilience := kObj.Spec.Resilience; resilience == nil {
+		return false
+	}
+	if configuration := kObj.Spec.Resilience.CounterStorage; configuration != nil {
+		return true
+	}
+	return false
+}
+
+func (r *ResilienceCounterStorageReconciler) updateLimitador(ctx context.Context, lObj *limitadorv1alpha1.Limitador) error {
+	obj, err := controller.Destruct(lObj)
+	if err != nil {
+		return err
+	}
+	_, err = r.Client.Resource(kuadrantv1beta1.LimitadorsResource).Namespace(lObj.GetNamespace()).Update(ctx, obj, metav1.UpdateOptions{})
+	return err
+}
+
+// INFO: Postconditon Section
+
+func NewResilienceDeploymentPostcondition() *ResilienceDeploymentPostcondition {
+	return &ResilienceDeploymentPostcondition{}
+}
+
+type ResilienceDeploymentPostcondition struct{}
+
+func (r *ResilienceDeploymentPostcondition) Subscription() controller.Subscription {
+	return controller.Subscription{
+		ReconcileFunc: r.run,
+		Events: []controller.ResourceEventMatcher{
+			{Kind: &kuadrantv1beta1.KuadrantGroupKind},
+		},
+	}
+}
+
+func (r *ResilienceDeploymentPostcondition) run(ctx context.Context, _ []controller.ResourceEvent, _ *machinery.Topology, _ error, _ *sync.Map) error {
+	logger := controller.LoggerFromContext(ctx).WithName("ResilienceDeploymentPrecondition")
+
+	logger.V(level).Info("ResilienceDeployment Postcondition", "status", "started")
+	defer logger.V(level).Info("ResilienceDeployment Postcondition", "status", "completed")
+	return nil
+}
+
+// INFO: Local Functions
+
+func isExperimentalFeatureEnabled(topology *machinery.Topology) bool {
+	k := GetKuadrantFromTopology(topology)
+	if k == nil {
+		return false
+	}
+
+	if val, exists := k.GetAnnotations()[ResilienceFeatureAnnotation]; exists {
```

Comment:
So, to enable this I have to define an annotation like this:

```yaml
annotations:
  kuadrant.io/experimental-dont-use-resilient-data-plane: true
```

Should the annotation name be changed?

Reply:
I am not sure if it needs to be changed. As soon as the all the work in the area is done the need for the annotation would be removed. This is most there so we can do iterative work in to main without affecting existing user. This is an undocumented feature.